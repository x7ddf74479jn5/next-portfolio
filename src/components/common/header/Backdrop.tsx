import styles from "src/styles/components/header/backdrop.module.scss";

type Props = {
  handleClick: () => void;
};

const Backdrop: React.VFC<Props> = ({ handleClick }) => {
  return (
    <button
      className={styles.backdrop}
      id="js-black-bg"
      onKeyDown={(e) => {
        return;
      }}
      onClick={() => handleClick()}
    ></button>
  );
};

export default Backdrop;
