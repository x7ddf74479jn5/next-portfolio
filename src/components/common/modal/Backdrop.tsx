import styles from "src/styles/components/modal/backdrop.module.scss";

type Props = {
  handleClick: () => void;
};

const Backdrop: React.VFC<Props> = ({ handleClick }) => {
  return (
    <button
      className={styles.backdrop}
      id="js-black-bg"
      onKeyDown={() => handleClick()}
      onClick={() => handleClick()}
    ></button>
  );
};

export default Backdrop;
