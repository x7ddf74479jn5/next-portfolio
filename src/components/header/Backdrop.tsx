import styles from "src/styles/components/header/backdrop.module.scss";

type Props = {
  handleClick: () => void;
};

const Backdrop: React.VFC<Props> = ({ handleClick }) => {
  return (
    <button
      className={styles.backdrop}
      id="js-black-bg"
      onClick={() => {
        return handleClick();
      }}
    ></button>
  );
};

export default Backdrop;
