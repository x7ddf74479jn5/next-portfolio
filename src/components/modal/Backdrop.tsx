import clsx from "clsx";
import styles from "src/styles/components/modal/Backdrop.module.scss";

type Props = {
  opacity?: 10 | 70;
};

const Backdrop: React.VFC<Props> = ({ opacity = 10 }) => {
  return (
    <div
      className={clsx({
        [styles.backdrop_70]: opacity === 70,
        [styles.backdrop]: opacity === 10,
      })}
    ></div>
  );
};

export default Backdrop;
