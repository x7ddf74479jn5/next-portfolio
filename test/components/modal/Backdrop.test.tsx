import Backdrop from "src/components/modal/Backdrop";
import type { RenderResult } from "test/test-utils";
import { render } from "test/test-utils";

describe("Backdrop", () => {
  let renderResult: RenderResult;

  describe("default", () => {
    beforeEach(() => {
      renderResult = render(<Backdrop />);
    });

    it("matches snapshot", () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      expect(renderResult.container.firstElementChild).toHaveClass("backdrop");
    });
  });

  describe("opacity 10", () => {
    beforeEach(() => {
      renderResult = render(<Backdrop opacity={10} />);
    });

    it("matches snapshot", () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      expect(renderResult.container.firstElementChild).toHaveClass("backdrop");
    });
  });

  describe("opacity 10", () => {
    beforeEach(() => {
      renderResult = render(<Backdrop opacity={70} />);
    });

    it("matches snapshot", () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      expect(renderResult.container.firstElementChild).toHaveClass("backdrop_70");
    });
  });
});
