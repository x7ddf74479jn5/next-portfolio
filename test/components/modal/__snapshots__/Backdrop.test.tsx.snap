// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop default matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="backdrop"
  />
</DocumentFragment>
`;

exports[`Backdrop opacity 10 matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="backdrop"
  />
</DocumentFragment>
`;

exports[`Backdrop opacity 10 matches snapshot 2`] = `
<DocumentFragment>
  <div
    class="backdrop_70"
  />
</DocumentFragment>
`;
